### PR TITLE
TST: filter warnings for is_extension_type deprecation

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Follow up on https://github.com/pandas-dev/pandas/pull/29457

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
cc @jbrockmendel

File: pandas/tests/dtypes/test_common.py [hunk 1]
```diff
@@ -573,6 +574,35 @@ def test_is_extension_type(check_scipy):
...
 
+def test_is_extension_type_deprecation():
+    with tm.assert_produces_warning(FutureWarning):
+        com.is_extension_type([1, 2, 3])
+
+
+@pytest.mark.parametrize(
+    "check_scipy", [False, pytest.param(True, marks=td.skip_if_no_scipy)]
+)
+def test_is_extension_array_dtype(check_scipy):
```
Comment (jorisvandenbossche) [inline pandas/tests/dtypes/test_common.py]: I copy pasted the original `is_extension_type` to test `is_extension_array_dtype`, as that is going to replace it and was not yet tested here (so when cleaning this up, can simply remove the above `is_extension_type` test)

**Comment by jbrockmendel** [comment]
I'm surprised we dont have better tests for is_extension_array_dtype.  LGTM.

**Review by TomAugspurger — APPROVED** [review]
> I'm surprised we dont have better tests for is_extension_array_dtype.

What's this referring to?

**Comment by jbrockmendel** [comment]
> What's this referring to?

the need for `test_is_extension_array_dtype` this implements

**Comment by TomAugspurger** [comment]
I assume it's verifying that `is_extension_array_dtype` is indeed a drop-in replacement for all the (tested) cases of `is_extension_dtype`. Some are probably duplicative.

**Comment by jorisvandenbossche** [comment]
> I assume it's verifying that is_extension_array_dtype is indeed a drop-in replacement for all the (tested) cases of is_extension_dtype.

Yes, but Brock's comment was about that this was needed, because `is_extension_array_dtype` did not yet have its own dedicated tests in `test_common.py` (it has some indirect tests in the extension array tests)

**Pull request merged**